### DISCUSSION TITLE
[Question] Budget path when using lighthouse programatically

### DISCUSSION THREAD

**Discussion by mfrachet** [GoogleChrome/lighthouse#12897] [Q&A]
Hey folks 👋🏻 

Thank you for the amazing job you do with this module 🙏🏻 .

I'm building a module relying on lighthouse and I'm wondering if you plan to add the budget-path option to the node (programatic) API?

Thank you 😊

**Comment by wangdahoo** [comment]
+1

**Comment by brendankenny** [comment]
There's not currently a plan to add `budget-path` to the node API. Generally we've preferred things like loading configuration from a file to stay in the CLI while the node API would be configured with an options object directly. Admittedly we don't always follow this and the config format can make this a little bit awkward, though.

To use budgets programmatically, it would be something like

```js
const lighthouse = require('lighthouse');

const runnerResult = await lighthouse('https://example.com', {logLevel: 'info'}, {
  extends: 'lighthouse:default',
  settings: {
    budgets: [{
      path: '/',
      resourceSizes: [{
        resourceType: 'image',
        budget: 1000,
      }],
    }],
  },
});
```

(the config `budgets` property is the exact [same format](https://github.com/GoogleChrome/lighthouse/blob/fc3907ff4945aa9f74b180ba151aff188fac5c53/types/budget.d.ts) as the budget file format, so it could also be read from file and set as a property on the config object, which is [all the CLI does](https://github.com/GoogleChrome/lighthouse/blob/3aa15dc6990bdfbb963d423267ff2317257fe338/lighthouse-cli/bin.js#L73-L78) to load a budget from a path, no extra processing going on)

As I said, it is a little awkward between the `extends: 'lighthouse:default'` and the fact that it's under `config.settings`, not a top level property, so maybe it would be worth moving `budgetPath` to [`LH.Flags`](https://github.com/GoogleChrome/lighthouse/blob/fc3907ff4945aa9f74b180ba151aff188fac5c53/types/externs.d.ts#L215-L230).
